File: 3.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        max_l = 1 if len(s) else 0
        ss = []
        for a in s:
            if not a in ss:
                ss.append(a)
            else:
                l = len(ss)
                if l>max_l:
                    max_l = l
                ss = ss[ss.index(a)+1:]
                ss.append(a)
        
        l = len(ss)
        if l>max_l:
            max_l = l
                    
        return max_l

File: 3.cpp
#include <string>
#include <vector>

using namespace std;
class Solution {
public:
  int lengthOfLongestSubstring(string s) {
    int max_length = 0;
    vector<char> ss;
    for (auto chr : s) {
      auto it = find(ss.begin(), ss.end(), chr);
      if (it == ss.end()) {
        ss.push_back(chr);
      } else {
        max_length = max(max_length, int(ss.size()));
        ss = vector<char>(it + 1, ss.end());
        ss.push_back(chr);
      }
    }
    max_length = max(max_length, int(ss.size()));
    return max_length;
  }
};